### DISCUSSION TITLE
Installation Troubles

### DISCUSSION THREAD

**Discussion by fjaiv** [Blair2004/NexoPOS#1472] [General]
Hi All, 
I'm having a pretty rough time installing NexoPOS.  I've tried both in Ubuntu 22.04, 20.04 and Win10.  So far, i've had the most luck in Win10, so i'll go from there. 

I followed the Youtube video below, using laragon for the install.  Most versions of software are, however newer than in the video. 
https://www.youtube.com/watch?v=-eXapKZrcBc&t=1s&ab_channel=NexoPOS

What i'm finding is that when I get to the Create Installation screen, I click "create installation", and eventually I see a network error at the bottom of the page.  I can verify that the DB was created and all tables are there.  Reloading the page brought me to a database update - after running for a bit, i'm greeted with an error that says "Attempt to read property "id" on null". I can "try again" or "return".  Try again results in the same ending. 

Can someone perhaps point out what I may be doing wrong?   I'd really prefer to run this under ubuntu 22.04, but all attempts there eventually result in a blank page being served, or a 404 after clicking 'dashboard' on the opening page instead of going to db setup.

**Comment by Blair2004** [comment]
Are you running "php artisan serve" ?

**Reply by fjaiv** [reply]
yes sir.   Thank you for the response! 

I do see a lot of activity in the php window while it's running - but no errors.

**Reply by Blair2004** [reply]
This is no longer a proper way of using NexoPOS. We have many guides on how to install NexoPOS on a real server.

https://my.nexopos.com/en/documentation

**Reply by fjaiv** [reply]
Thank Blair.  I did try using that documentation, but it only outlines ubuntu 20.04 not 22.04.   I actually used that guide three times, starting from a fresh ubuntu 20.04 install each time.  I end up running into problems with php7 vs php8.1 which ultimately ended with either a 404 after clicking the initial dashboard link, or blank pages being served :(

**Reply by fjaiv** [reply]
so i attempted to use the official documentation again.  problems that I run into:

When I get to the part where we install phpmyadmin, dependencies and extensions are installed for PHP8.2 instead of 8.1  This results in phpmyadmin not working, and the nexopos site not working.  So I uninstall phpmyadmin and it's dependencies, and all of the 8.2 modules/extensions.  reinstall phpmyadmin from .zip file.  After this, I can get nexopos and phpmyadmin site to load, however whenever I click "dashboard" "Sign in" or "sign up" I end up with a 404 error.

**Reply by Blair2004** [reply]
You can run different PHP version for NexoPOS and phpmyadmin. If you're using a remote server I suggest you using a tool like Laravel Forge that helps setup project such as NexoPOS in no time.

**Reply by nzian** [reply]
I am also a windows 10 user and wsl2 ubuntu 20.04. I actually use nexopos from php 7.4 and apache2 and so when laravel version updated I will move to php8.1 and nginx server. I use virtual host setup in the server and also run in both dev and prod version perfectly. If you have need nginx or apache virtual host config file let me know @fjaiv 
Thank you

**Reply by fjaiv** [reply]
thank you both for the help.  I will keep trying under ubuntu 22.04.

**Comment by georgebastion** [comment]
Did you get a fix for this error? I am getting a network error when completing the installation.

**Reply by Blair2004** [reply]
Hi, 
Refer to this comment https://github.com/Blair2004/NexoPOS/discussions/1472#discussioncomment-5815118

**Comment by kabujeh** [comment]
I am getting this error when running composer install. Class UseThisChain located in ./app/Services/HelperFunctions.php does not comply with psr-4 autoloading standard (rule: App\ => ./app). Skipping. Am using larvel with php8.2. I cant seem to resolve this error. Kindly assist.

**Reply by Blair2004** [reply]
Hi, are you still having that issue ?

**Comment by kabujeh** [comment]
Yes. I wasn't able to sort it out

**Reply by Blair2004** [reply]
So you're having this issue with v5.3.3?
